File: Components/FAQsInfo.vb
'
' DotNetNuke® - http://www.dotnetnuke.com
' Copyright (c) 2002-2006
' by Perpetual Motion Interactive Systems Inc. ( http://www.perpetualmotion.ca )
'
' Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
' documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
' the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
' to permit persons to whom the Software is furnished to do so, subject to the following conditions:
'
' The above copyright notice and this permission notice shall be included in all copies or substantial portions 
' of the Software.
'
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
' THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
' CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
' DEALINGS IN THE SOFTWARE.
'

Imports System

Namespace DotNetNuke.Modules.FAQs

    ''' <summary>
    ''' Info class for FAQs
    ''' </summary>
    Public Class FAQsInfo

#Region "Private Members"
        Private _itemId As Integer
        Private _moduleId As Integer
        Private _categoryId As Integer
        Private _question As String
        Private _answer As String
        Private _createdByUser As String
        Private _createdByUserName As String
        Private _createdDate As DateTime
        Private _dateModified As DateTime
        Private _viewCount As Integer
        Private _faqCategoryName As String
        Private _faqCategoryDescription As String
        Private _index As Integer
#End Region

#Region "Constructors"
        Public Sub New()
        End Sub

        ''' <summary>
        ''' Initializes a new instance of the <see cref="FAQsInfo" /> class.
        ''' </summary>
        ''' <param name="itemId">The item id.</param>
        ''' <param name="moduleId">The module id.</param>
        ''' <param name="categoryId">The category id.</param>
        ''' <param name="questionTitle">The question title.</param>
        ''' <param name="question">The question.</param>
        ''' <param name="answer">The answer.</param>
        ''' <param name="createdByUser">The created by user.</param>
        ''' <param name="createdDate">The created date.</param>
        ''' <param name="dateModified">The date modified.</param>
        ''' <param name="viewCount">The view count.</param>
        Public Sub New(ByVal itemId As Integer, ByVal moduleId As Integer, ByVal categoryId As Integer, ByVal questionTitle As String, ByVal question As String, ByVal answer As String, ByVal createdByUser As String, ByVal createdDate As DateTime, ByVal dateModified As DateTime, ByVal viewCount As Integer)
            Me.ItemId = itemId
            Me.ModuleId = moduleId
            Me.CategoryId = categoryId
            Me.Question = question
            Me.Answer = answer
            Me.CreatedByUser = createdByUser
            Me.CreatedDate = createdDate
            Me.DateModified = dateModified
            Me.ViewCount = viewCount
        End Sub
#End Region

#Region "Public Properties"

        ''' <summary>
        ''' Gets or sets the item id.
        ''' </summary>
        ''' <value>The item id.</value>
        Public Property ItemId() As Integer
            Get
                Return _itemId
            End Get
            Set(ByVal Value As Integer)
                _itemId = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the name of the FAQ category.
        ''' </summary>
        ''' <value>The name of the FAQ category.</value>
        Public Property FaqCategoryName() As String
            Get
                Return _faqCategoryName
            End Get
            Set(ByVal Value As String)
                _faqCategoryName = Value
            End Set
        End Property
        
        ''' <summary>
        ''' Gets or sets the FAQ category description.
        ''' </summary>
        ''' <value>The FAQ category description.</value>
        Public Property FaqCategoryDescription() As String
            Get
                Return _faqCategoryDescription
            End Get
            Set(ByVal Value As String)
                _faqCategoryDescription = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the module id.
        ''' </summary>
        ''' <value>The module id.</value>
        Public Property ModuleId() As Integer
            Get
                Return _moduleId
            End Get
            Set(ByVal Value As Integer)
                _moduleId = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the category id.
        ''' </summary>
        ''' <value>The category id.</value>
        Public Property CategoryId() As Integer
            Get
                Return _categoryId
            End Get
            Set(ByVal Value As Integer)
                _categoryId = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the question.
        ''' </summary>
        ''' <value>The question.</value>
        Public Property Question() As String
            Get
                Return _question
            End Get
            Set(ByVal Value As String)
                _question = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the answer.
        ''' </summary>
        ''' <value>The answer.</value>
        Public Property Answer() As String
            Get
                Return _answer
            End Get
            Set(ByVal Value As String)
                _answer = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the created by user.
        ''' </summary>
        ''' <value>The created by user.</value>
        Public Property CreatedByUser() As String
            Get
                Return _createdByUser
            End Get
            Set(ByVal Value As String)
                _createdByUser = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the name of the created by user.
        ''' </summary>
        ''' <value>The name of the created by user.</value>
        Public Property CreatedByUserName() As String
            Get
                Return _createdByUserName
            End Get
            Set(ByVal Value As String)
                _createdByUserName = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the created date.
        ''' </summary>
        ''' <value>The created date.</value>
        Public Property CreatedDate() As DateTime
            Get
                Return _createdDate
            End Get
            Set(ByVal Value As DateTime)
                _createdDate = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the date modified.
        ''' </summary>
        ''' <value>The date modified.</value>
        Public Property DateModified() As DateTime
            Get
                Return _dateModified
            End Get
            Set(ByVal Value As DateTime)
                _dateModified = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the view count.
        ''' </summary>
        ''' <value>The view count.</value>
        Public Property ViewCount() As Integer
            Get
                Return _viewCount
            End Get
            Set(ByVal Value As Integer)
                _viewCount = Value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the index.
        ''' </summary>
        ''' <value>The index.</value>
        Public Property Index() As Integer
            Get
                Return _index
            End Get
            Set(ByVal Value As Integer)
                _index = Value
            End Set
        End Property

#End Region

    End Class

End Namespace

File: Components/FAQsInfo.cs
//
// DotNetNuke® - http://www.dotnetnuke.com
// Copyright (c) 2002-2011
// by DotNetNuke Corporation
//
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all copies or substantial portions
// of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
// DEALINGS IN THE SOFTWARE.
//

using System;
using System.Data;
using DotNetNuke.Common.Utilities;
using DotNetNuke.Entities.Modules;

namespace DotNetNuke.Modules.FAQs
{
	
	/// <summary>
	/// Info class for FAQs
	/// </summary>
	[Serializable()]
	public class FAQsInfo:IHydratable
	{
		#region Private Members

		private int _itemId;
		private int _moduleId;
		private int _categoryId;
		private string _question;
		private string _answer;
		private string _createdByUser;
		private string _createdByUserName;
		private DateTime _createdDate;
		private DateTime _dateModified;
		private int _viewCount;
		private string _faqCategoryName;
		private string _faqCategoryDescription;
		private int _index;
		
		#endregion
		
		#region Constructors
		
		public FAQsInfo()
		{
		}
		
		/// <summary>
		/// Initializes a new instance of the <see cref="FAQsInfo" /> class.
		/// </summary>
		/// <param name="itemId">The item id.</param>
		/// <param name="moduleId">The module id.</param>
		/// <param name="categoryId">The category id.</param>
		/// <param name="question">The question.</param>
		/// <param name="answer">The answer.</param>
		/// <param name="createdByUser">The created by user.</param>
		/// <param name="createdDate">The created date.</param>
		/// <param name="dateModified">The date modified.</param>
		/// <param name="viewCount">The view count.</param>
		public FAQsInfo(int itemId, int moduleId, int categoryId, string question, string answer, string createdByUser, DateTime createdDate, DateTime dateModified, int viewCount)
		{
			_itemId = itemId;
			_moduleId = moduleId;
			_categoryId = categoryId;
			_question = question;
			_answer = answer;
			_createdByUser = createdByUser;
			_createdDate = createdDate;
			_dateModified = dateModified;
			_viewCount = viewCount;
			_createdByUserName = "";
			_faqCategoryDescription = "";
			_faqCategoryName = "";
			_index = 0;
		}
		#endregion
		
		#region Public Properties
		
		/// <summary>
		/// Gets or sets the item id.
		/// </summary>
		/// <value>The item id.</value>
		public int ItemId
		{
			get
			{
				return _itemId;
			}
			set
			{
				_itemId = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the name of the FAQ category.
		/// </summary>
		/// <value>The name of the FAQ category.</value>
		public string FaqCategoryName
		{
			get
			{
				return _faqCategoryName;
			}
			set
			{
				_faqCategoryName = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the FAQ category description.
		/// </summary>
		/// <value>The FAQ category description.</value>
		public string FaqCategoryDescription
		{
			get
			{
				return _faqCategoryDescription;
			}
			set
			{
				_faqCategoryDescription = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the module id.
		/// </summary>
		/// <value>The module id.</value>
		public int ModuleId
		{
			get
			{
				return _moduleId;
			}
			set
			{
				_moduleId = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the category id.
		/// </summary>
		/// <value>The category id.</value>
		public int CategoryId
		{
			get
			{
				return _categoryId;
			}
			set
			{
				_categoryId = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the question.
		/// </summary>
		/// <value>The question.</value>
		public string Question
		{
			get
			{
				return _question;
			}
			set
			{
				_question = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the answer.
		/// </summary>
		/// <value>The answer.</value>
		public string Answer
		{
			get
			{
				return _answer;
			}
			set
			{
				_answer = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the created by user.
		/// </summary>
		/// <value>The created by user.</value>
		public string CreatedByUser
		{
			get
			{
				return _createdByUser;
			}
			set
			{
				_createdByUser = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the name of the created by user.
		/// </summary>
		/// <value>The name of the created by user.</value>
		public string CreatedByUserName
		{
			get
			{
				return _createdByUserName;
			}
			set
			{
				_createdByUserName = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the created date.
		/// </summary>
		/// <value>The created date.</value>
		public DateTime CreatedDate
		{
			get
			{
				return _createdDate;
			}
			set
			{
				_createdDate = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the date modified.
		/// </summary>
		/// <value>The date modified.</value>
		public DateTime DateModified
		{
			get
			{
				return _dateModified;
			}
			set
			{
				_dateModified = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the view count.
		/// </summary>
		/// <value>The view count.</value>
		public int ViewCount
		{
			get
			{
				return _viewCount;
			}
			set
			{
				_viewCount = value;
			}
		}
		
		/// <summary>
		/// Gets or sets the index.
		/// </summary>
		/// <value>The index.</value>
		public int Index
		{
			get
			{
				return _index;
			}
			set
			{
				_index = value;
			}
		}
		
		#endregion

		#region Implementation of IHydratable

		public void Fill(IDataReader dr)
		{
			_itemId = Null.SetNullInteger(dr["ItemId"]);
			_moduleId = Null.SetNullInteger(dr["ModuleId"]);
			_categoryId = Null.SetNullInteger(dr["CategoryId"]);
			_question = Null.SetNullString(dr["Question"]);
			_answer = Null.SetNullString(dr["Answer"]);
			_createdByUserName = Null.SetNullString(dr["CreatedByUserName"]);
			_createdDate = Null.SetNullDateTime(dr["CreatedDate"]);
			_dateModified = Null.SetNullDateTime(dr["DateModified"]);
			_viewCount = Null.SetNullInteger(dr["ViewCount"]);
			_faqCategoryName = Null.SetNullString(dr["FaqCategoryName"]);
			_faqCategoryDescription = Null.SetNullString(dr["FaqCategoryDescription"]);
		}

		public int KeyID
		{
			get { return _itemId; }
			set { _itemId = value; }
		}

		#endregion
	}
	
}

